### PR TITLE
api: Rename `StreamMessage` to `ChannelMessage`

### PR DESCRIPTION
**Pull request by Khader-1**
Fixes parts of #631

### CODE REVIEW

**Review by sm-sayedi — APPROVED** [review]
Thanks @Khader-1. LGTM. Moving on to the integration review.

File: test/example_data.dart [hunk 1]
```diff
@@ -302,22 +302,22 @@ Map<String, dynamic> _messagePropertiesFromContent(String? content, String? cont
 int _nextMessageId() => (_lastMessageId += 1 + Random().nextInt(100));
 int _lastMessageId = 1000;
 
-const defaultStreamMessageStreamId = 123;
+const defaultChannelMessageStreamId = 123;
 
-/// Construct an example stream message.
+/// Construct an example stream channel.
```
Comment (gnprice) [inline test/example_data.dart:307]: "channel message"

File: test/example_data.dart [hunk 2]
```diff
@@ -302,22 +302,22 @@ Map<String, dynamic> _messagePropertiesFromContent(String? content, String? cont
...
+/// calls to [channelMessage] and [dmMessage].
 ///
 /// The message will be in `stream` if given.  Otherwise,
-/// an example stream with ID `defaultStreamMessageStreamId` will be used.
+/// an example stream with ID `defaultChannelMessageStreamId` will be used.
 ///
 /// See also:
 ///  * [dmMessage], to construct an example direct message.
-StreamMessage streamMessage({
+ChannelMessage channelMessage({
```
Comment (gnprice) [inline test/example_data.dart:320]: Because this function `eg.streamMessage` has a huge number of references (333 of them, says my IDE), its rename should go in its own commit, separate from editing other things that refer to `StreamMessage` or "stream messages".

That'd help make the changes easier to review — including self-review to catch things like "example stream channel" noted above.

**Review by gnprice — COMMENTED** [review]
Thanks @Khader-1! Comments below.

I haven't read through the whole second commit in detail; one comment is about how it can be made easier to review.

**Comment by gnprice** [comment]
Closing this PR as part of pushing back the priority of the remaining portion of this issue #631: details at https://github.com/zulip/zulip-flutter/issues/631#issuecomment-2266293080. Thanks again!
